compare-bug-version-and-git-branch.sh

```diff
@@ -19,21 +19,24 @@ if ! git show --pretty=format:%B | head -n 2 | tail -n 1 | egrep '^$' >/dev/null
     exit 1
 fi
 
-BUG=$(git show --name-only --format=email | awk '{IGNORECASE=1} /^BUG:/{print $2}' | tail -1)
-if [ -z "${BUG}" -a "${GERRIT_TOPIC}" = "rfc" ]; then
-    echo "No BUG id for rfc needed."
-    exit 0
-elif [ -a "${GERRIT_TOPIC#*rfc*}" != "${GERRIT_TOPIC}" -a "${GERRIT_BRANCH}" != "master" ]; then
-    echo "No BUG id for rfc needed in 'non-master' branch"
-    exit 0
-elif [ -z "${BUG}" ]; then
-    echo "No BUG id, but topic '${GERRIT_TOPIC}' does not match 'rfc'."
+REF=$(git show --name-only --format=email | awk '{IGNORECASE=1} /^bug: /{print $2}' | tail -1)
+if [ -z "${REF}" ] ; then
+    REF=$(git show --name-only --format=email | awk '{IGNORECASE=1} /^fixes: /{print $2}' | tail -1)
+fi
+if [ -z "${REF}" ] ; then
+    REF=$(git show --name-only --format=email | awk '{IGNORECASE=1} /^updates: /{print $2}' | tail -1)
+fi
+if [ -z "${REF}" ]; then
+    echo "No reference id for the patch"
```

Comment:
Let's make this a better string, more like a readme, such that first time failure in smoke has detail on how to fix this and resubmit. Would be essential for first time contributors to understand this more easily.

So that more text is shown in jenkins logs: https://build.gluster.org/job/compare-bug-version-and-git-branch/30758/console (example)

It would be better if we could post this against the commit itself as a review comment.

Reply:
Ack!